docs/ROOTFS_DISCOVERY.md

```diff
@@ -0,0 +1,395 @@
+   persistency of modifications of the file system is not available. If this
+   mode is used the URL to acquire the `rootfs` disk image from can be derived
+   automatically from the URL that was used to acquire the UKI itself. (This
+   information is passed from UKI to userspace via the `LoaderDevicePartURL`
+   EFI variable.)
+
+Much as for the the disk-based boot scheme described in the previous section
+discovery of the boot source can take place fully automatically, with each
+component taking the boot source choice of the preceeding component into
+account: the boot loader can automatically download UKIs from the same source
+itself was downloaded from. Moreover the `initrd` of the UKI can automatically
+downloads the `rootfs` from the same source itself was downloaded from.
+
+Also, much like in the disk-based boot scheme it is also possible to
+specifically direct each of the components to a different place instead of the
+automatic derivation of URLs from the same source. On top of that it is of
+course possible to mix disk-based and network-based boot: for example place the
+boot loader on the local disk, but use UKIs and `rootfs` from networked souces;
+or alternatively place both boot loader and UKIs on the local disk, and only
+the `rootfs` on a networked source.
+
+## Trust & Security
+
+In a modern boot integrity world, all three of the relevant components as well
+as (most of) their sidecars require cryptographic protection before they can be
+consumed. Specifically:
+
+1. The boot loader is typically authenticated by the firmware before invocation
+   via UEFI Secure Boot, i.e. checked against a cryptographic certificate list
+   persistently stored in the firmware. Note that the various auxiliary
+   resources available the `systemd-boot` boot loader reads are not
+   individually authenticated (i.e. `loader.conf` as well as Type #1 Boot
+   Loader Specification entries). Because of this they can typically only be
+   used in a very restricted fashion, i.e. configure some UI details as well as
+   menu entries. Some options available are ignored if Secure Boot mode is
+   enabled. Moreover, even if the text strings shown in the menu entries might
+   not be authenticated the binaries that are invoked once they are selected
+   are.
+
+2. The UKIs are also authenticated by the firmware via UEFI Secure Boot, and so
+   are EFI Addons. `confext` and `sysext` sidecars are protected via
+   `dm-verity` and a signature of the root hash is validated against keys in
+   the kernel's built-in keyring. System credentials are authenticated via
+   secrets stored in the TPM.
+
+3. Authentication of the `rootfs` and `usrfs` is more variable: depending on
+   setup this is either done via `dm-verity` (either pinned by root hash from
+   the UKI, or authenticated by signature prvoided to kernel, checked against
+   the kernel's built-in keyring) or `dm-crypt`+`dm-integrity` (protected by
```

Comment:
not just the built-in keyring (which is assembled at kernel compile time) but also platform and machine keyrings, which are derived from uefi db/mok. So maybe just say "keyrings" plural if you don't want to specify all the details.

Reply:
Does anyone actually layer `dm-crypt` on top of `dm-integrity` in practice?

Reply:
> Does anyone actually layer `dm-crypt` on top of `dm-integrity` in practice?

I do not at least, but don't tell anyone. But then again, people sent us patches to make it work hence I figure some people do...